include/oneapi/dpl/internal/async_impl/async_utils.h

```diff
@@ -96,12 +104,19 @@ class __future<_T, typename std::enable_if<__par_backend_hetero::__internal::is_
             __tmp = ::std::unique_ptr<__par_backend_hetero::__lifetime_keeper<_Ts...>>(
                 new __par_backend_hetero::__lifetime_keeper<_Ts...>(__t...));
     }
+    __future(_T __d) : __par_backend_hetero::__future_base(sycl::event{}), __data(__d) {}
     _T
     get()
     {
         this->wait();
         return __data;
     }
+    __future(__par_backend_hetero::__future<typename ::std::iterator_traits<_T>::value_type>&& __o, _T __d)
```

Comment:
Could you please add a comment why we need this ctor?

Reply:
Done, Thanks for you feedback. I've cleaned up the ctor's of our future implementation and added comments. Please have a second look.